### PR TITLE
Add Weekly job for model analysis

### LINKED ISSUE #662: [Model Analysis] Weekly job that will automatically update model and op details
Based on existing scripts, create a weekly job that will generate all markdowns and update them automatically. This one should be run over weekends.

For reference, we already have a sample of doing so, just for the FFE/MLIR uplifts. We can utilize sam structure to generate and run these.

### PR DESCRIPTION
**Pull request by chandrasekaranpradeep**
Fixes #662 
1. Add new marker `model_analysis` for the models inference test present in the forge/test/models/pytorch path
2. Modified the script to collect the test based upon the marker `model_analysis`
3. Created a workflow(i.e model-analysis-weekly.yml) for running the model analysis script  which will generate markdown files and then create a PR for updating the model_analysis_docs(Markdown files).
4. Moved the build-image job in Build and Test workflow(i.e build-and-test.yml) to separate workflow, so that I can reuse the workflow for the Model Analysis Weekly workflow(i.e model-analysis-weekly.yml) 

Note: The workflow job will trigger every week on Friday at 11:00 PM UTC (Saturday at 12:00 AM Serbia time).

### CODE REVIEW

File: .github/workflows/model-analysis-weekly.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+          -DCMAKE_CXX_COMPILER=clang++ \
+          -DCMAKE_C_COMPILER_LAUNCHER=ccache \
+          -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
+          cmake --build ${{ steps.strings.outputs.build-output-dir }}
+
+      - name: Run Model Analysis Script
+        shell: bash
+        run: |
+          source env/activate
+          apt install -y libgl1-mesa-glx
```
Comment (milosevic81) [inline .github/workflows/model-analysis-weekly.yml:75]: You may want to set "set pipefail" here, otherwise it may report a false positive result

Comment (chandrasekaranpradeep) [reply .github/workflows/model-analysis-weekly.yml:75]: Sure Will update it

**Review by vmilosevic — APPROVED** [review]

File: .github/workflows/model-analysis-weekly.yml (hunk 1 above)
Comment (chandrasekaranpradeep) [reply .github/workflows/model-analysis-weekly.yml:75]: Added the `set -o pipefail ` command to change the pipeline exit status to the first command that fail.

**Review by dgolubovicTT — APPROVED** [review]

File: forge/test/conftest.py [hunk 2]
```diff
@@ -448,3 +448,23 @@ def pytest_runtest_logreport(report):
         for key, default_value in environ_before_test.items():
             if os.environ.get(key, "") != default_value:
                 os.environ[key] = default_value
+
+
+def pytest_collection_modifyitems(config, items):
+
+    marker = config.getoption("-m")  # Get the marker from the -m option
+
+    if marker and marker == "model_analysis":  # If a marker is specified
```
Comment (vmilosevic) [inline forge/test/conftest.py:457]: Please check if this is correct, marker can also be an expression and "not model_analysis" or "push and nightly" are all valid -m options, how will 
```if marker and marker == "model_analysis```
work in that case?

**Pull request merged**